app/models/test_run.rb

```diff
@@ -9,6 +9,24 @@ class TestRun < ApplicationRecord
   validates_numericality_of :time_to_service_estimate, greater_than_or_equal_to: 0, only_integer: true, allow_nil: true
   validates_numericality_of :time_to_service, greater_than_or_equal_to: 0, only_integer: true, allow_nil: true
 
+  STATUSES = {
+    complete: 'complete',
+    in_progress: 'in_progress',
+    cancelled: 'cancelled'
+  }.freeze
+
+  def status
+    return STATUSES[:complete] unless test_script_results.empty?
+    return STATUSES[:cancelled] if time_to_service&.zero?
```

Comment:
I'm not sure about using zero instead of `nil` here. Is the server guaranteed never to send back 0 for the estimated time to completion?

Reply:
Mmm you are right, it should send 0 if it's going to be served immediately.

Reply:
+1 (or maybe -1) for the negative value, a new column would be overkill